roles/glance/templates/etc/glance/glance-api-paste.ini

```diff
@@ -1,50 +1,50 @@
+pipeline = healthcheck versionnegotiation osprofiler http_proxy_to_wsgi unauthenticated-context cache rootapp
 
 # Use this pipeline for caching w/ management interface but no auth
 [pipeline:glance-api-cachemanagement]
-pipeline = healthcheck versionnegotiation osprofiler unauthenticated-context cache cachemanage rootapp
+pipeline = healthcheck versionnegotiation osprofiler http_proxy_to_wsgi unauthenticated-context cache cachemanage rootapp
 
 # Use this pipeline for keystone auth
 [pipeline:glance-api-keystone]
 {% if glance.auditing.enabled|bool %}
-pipeline = healthcheck versionnegotiation osprofiler authtoken audit context rootapp
+pipeline = healthcheck versionnegotiation osprofiler http_proxy_to_wsgi authtoken audit context rootapp
 {% else %}
-pipeline = healthcheck versionnegotiation osprofiler authtoken context rootapp
+pipeline = healthcheck versionnegotiation osprofiler http_proxy_to_wsgi authtoken context rootapp
 {% endif %}
 
 # Use this pipeline for keystone auth with image caching
 [pipeline:glance-api-keystone+caching]
 {% if glance.auditing.enabled|bool %}
-pipeline = healthcheck versionnegotiation osprofiler authtoken audit context cache rootapp
+pipeline = healthcheck versionnegotiation osprofiler http_proxy_to_wsgi authtoken audit context cache rootapp
 {% else %}
-pipeline = healthcheck versionnegotiation osprofiler authtoken context cache rootapp
+pipeline = healthcheck versionnegotiation osprofiler http_proxy_to_wsgi authtoken context cache rootapp
 {% endif %}
 
 # Use this pipeline for keystone auth with caching audit and cache management
 [pipeline:glance-api-keystone+cachemanagement]
 {% if glance.auditing.enabled|bool %}
-pipeline = healthcheck versionnegotiation osprofiler authtoken audit context cache cachemanage rootapp
+pipeline = healthcheck versionnegotiation osprofiler http_proxy_to_wsgi authtoken audit context cache cachemanage rootapp
 {% else %}
-pipeline = healthcheck versionnegotiation osprofiler authtoken context cache cachemanage rootapp
+pipeline = healthcheck versionnegotiation osprofiler http_proxy_to_wsgi authtoken context cache cachemanage rootapp
 {% endif %}
 
 # Use this pipeline for authZ only. This means that the registry will treat a
 # user as authenticated without making requests to keystone to reauthenticate
 # the user.
 [pipeline:glance-api-trusted-auth]
-pipeline = healthcheck versionnegotiation osprofiler context rootapp
+pipeline = healthcheck versionnegotiation osprofiler http_proxy_to_wsgi context rootapp
 
 # Use this pipeline for authZ only. This means that the registry will treat a
 # user as authenticated without making requests to keystone to reauthenticate
 # the user and uses cache management
 [pipeline:glance-api-trusted-auth+cachemanagement]
-pipeline = healthcheck versionnegotiation osprofiler context cache cachemanage rootapp
+pipeline = healthcheck versionnegotiation osprofiler http_proxy_to_wsgi context cache cachemanage rootapp
```

Comment:
I think the proxy needs to go before the versionnegotiation, since I think that sends back links for each version.